src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -1003,12 +1012,71 @@ private async Task CommonStreaming(ConnectionState connectionState, string strea
         }
     }
 
+    private async Task<(ConnectionState, Activity?)> WaitForActiveConnectionWithActivityAsync(string sendingMethodName, string invokedMethodName, CancellationToken token)
+    {
+        // Start the activity before waiting on the connection.
+        // Starting the activity here means time to connect or reconnect is included in the invoke.
+        var activity = CreateActivity(invokedMethodName);
+
+        try
+        {
+            ConnectionState connectionState;
+            var connectionStateTask = _state.WaitForActiveConnectionAsync(sendingMethodName, token);
+            if (connectionStateTask.Status == TaskStatus.RanToCompletion)
```

Comment:
Do we need to set any server tags if `connectionStateTask` throws. It throws when you're either not connected or the user canceled the passed in CTS. In either case it's not actually doing anything except creating and stopping an activity.

Plus, it'd clean up the code a bit.

Reply:
If there is a problem starting the connection then it would be useful to know the server address being called

Reply:
But the connection failing to start, or closing before the send calls, or not even calling start in the first place, is not reflected in SendAsync or InvokeAsync calls. All it knows is that there is no connection.

Reply:
With telemetry it's useful to view it from the perspective of the user. The mechanics of what happens internally that could cause the call to fail are invisible to them. They're focused on the end result: the call they want to make failed and there was an error. Having information about what they were trying to do - make a call to a hub with the specified `rpc.service`, `rpc.method`, `server.address`, `server.port` - is what they care about. Then the error reason would be in `error.type`.

If you want to make it clearer that the call failed because of the connection not being available, there is room to do that by providing good values to `error.type`. By default, it is the exception type name, but it can be customized in known scenarios. For example, if there is a problem with the connection, it could be something like: `negotiate-failed`, `hub-connection-not-started`, etc.

Reply:
Also, `server.address` and `server.port` are considered required attributes according to the spec: https://github.com/open-telemetry/semantic-conventions/blob/1e34b57b9f73b08b109cdc0e8841e857e5f5c205/docs/rpc/rpc-spans.md#client-attributes

Reply:
Ok thanks, I wanted to know some concrete reason we were including it.